src/coreclr/vm/codeman.h

```diff
@@ -1912,6 +1912,16 @@ protected :
         return m_CPUCompileFlags;
     }
 
+#ifdef __GNUC__
+    __attribute__((target("sve")))
+#endif
+    inline UINT64 GetSystemVectorLength()
+    {
+        UINT64 size;
+        __asm__ __volatile__("cntb %0" : "=r"(size));
+        return size;
+    }
```

Comment:
I'd expect this to fail for MSVC, which doesn't allow inline assembly?

Can we just call `svcntb()` on Windows instead, or is there potentially an official OS API for this?

Reply:
I wasn't sure this was available for MSVC yet. If so, then great, that's easier.

Is there a min version of MSVC needed to build coreclr?

Reply:
Hmmm, it might not be available yet, at least its not available in the MSVC version CI is using. -- @kunalspathak might know better when that support is expected to land.

I expect we need some ifdef here regardless since `__asm` is explicitly unsupported on MSVC for anything but 32-bit x86 and for the general function to be configured to assert if called from Windows, as a safety measure. -- This function directly executing an SVE instruction unguarded is a bit dangerous and there isn't really a way to assert that it's being done safely. It's only safe in the current use-case since its only called if the relevant support was queried from the OS.

Reply:
> at least it's not available in the MSVC version CI is using

yes, I checked around and they haven't added that support yet, so it will not be anytime sooner

> Can we just call svcntb() on Windows instead

They do not and the guidance was to use whatever is exposed by the compiler, so we won't have it at least for new few months.

So, I think in short-term, let us do it in `asm` guarded with if CPU-capability supports SVE.

Reply:
The consideration is the msvc compiler does not support asm

Reply:
ah ok. Yes, we should probably move this to the `.asm` file so it can be used both by the runtime and jit (`getMaxTVectorByteLength()`

Reply:
Does `MASM` support SVE instructions yet? It may be in the same boat where if MSVC doesn't support it then it may not either.

Reply:
If it does not support it, you can always emit the instruction as hex code.

Reply:
Using ACLE on Linux and return a hardcode length (128-bits) on Windows.

Reply:
> Return a hardcoded vector length of 16-bytes until we find a suitable mechanism to retrieve it.

Just noting that we won't be able to actually ship in November in such a state. We will require some form of actual query of the hardware size from the OS to avoid any issues if its run on hardware that has larger length.

It might be overall simpler to just get such a fallback setup now so we don't need to worry about it, risk forgetting it about it, or anything along those lines.

Reply:
>It might be overall simpler to just get such a fallback setup now 

Agreed, but AIUI, for windows today:
* Can't use inline asm, including encoding in hex
* no OS API available
* no SVE ACLE available

I'm not aware of any other method.

Reply:
I think the solution is what @jkotas mentioned of emitting hex code or just write the assembly `cntb` in the `.asm` and `.S` file - https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/arm64/asmhelpers.asm.

Reply:
It looks like MASM v14.4 supports things and we can just define a function that does:
```asm
rdvl    x0, #1
ret
```

Reply:
Updated the PR to test how the ci handles use of `rdvl`.